### ISSUE TITLE
`numberToHex` is breaking chain IDs in 5.75.0

### ISSUE THREAD

**Issue by franjohn21** [thirdweb-dev/js#5652]
The `numberToHex` utility is returning an invalid chain hex, at least according to metamask. 

For example, trying to switch the chain to ethereum mainnet: `Expected 0x-prefixed, unpadded, non-zero hexadecimal string 'chainId'. Received:\n0x01`.

```
> console.log(numberToHex(1))
> 0x01
```
Should be `0x1`. The 0 padding is causing the above error at least in Metamask.

Broken on thirdweb@5.75.0
Works fine on thirdweb@5.74.0

**Comment by joaquim-verges** [comment]
Thank you for reporting this. Investigating

**Comment by joaquim-verges** [comment]
@franjohn21 just added this test, which passes fine

```
it("should convert number with no padding", () => {
    const result = numberToHex(1);
    expect(result).toBe("0x1");
  });
```

we also haven't changed this function from 5.75 to 5.76, but we did change the underlying implementation to use ox in https://github.com/thirdweb-dev/js/pull/5366 about a month ago.

is it possible that you're using a different version of ox in your project?

**Comment by gregfromstl** [comment]
Hey @franjohn21, this should be fixed with version 5.76.0
